gen/generate.go

```diff
@@ -309,22 +282,29 @@ func generateModule(m *compile.Module, i thriftPackageImporter, builder *generat
 			if _, err := builder.AddRootService(service); err != nil {
 				return nil, err
 			}
+		}
 
-			serviceFiles, err := Service(g, service)
-			if err != nil {
-				return nil, fmt.Errorf(
-					"could not generate code for service %q: %v",
-					serviceName, err)
-			}
-
-			if !o.NoServiceHelpers {
-				for name, buff := range serviceFiles {
-					files[name] = buff.Bytes()
-				}
-			}
+		err = Services(g, m.Services)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"could not generate code for services %v", err)
 		}
 	}
 
+	buff := new(bytes.Buffer)
+	if err := g.Write(buff, nil); err != nil {
+		return nil, fmt.Errorf(
+			"could not write code for services into %q: %v", o.SingleFile, err)
+	}
+
+	// The default is the package name
+	name := packageName + ".go"
+	if len(o.SingleFile) > 0 {
+		name = o.SingleFile
```

Comment:
Should decide on the output file name before including it in the error message above.